packages/@aws-cdk/aws-codepipeline/lib/pipeline.ts

```diff
@@ -255,7 +285,45 @@ export class Pipeline extends cdk.Construct implements cpapi.IPipeline {
 
   // ignore unused private method (it's actually used in Stage)
   // @ts-ignore
-  private _generateOutputArtifactName(stage: cpapi.IStage, action: cpapi.Action): string {
+  private _attachAction(stage: Stage, action: actions.Action): void {
+    // handle cross-region Actions here
+    if (action.region) {
+      // get the region the Pipeline itself is in
+      const pipelineStack = cdk.Stack.find(this);
+      const pipelineRegion = pipelineStack.requireRegion(
+        "You need to specify an explicit region when using CodePipeline's cross-region support");
+      const pipelineAccount = pipelineStack.requireAccountId(
+        "You need to specify an explicit account when using CodePipeline's cross-region support");
+      const app = pipelineStack.parent === undefined ? undefined : pipelineStack.parent as cdk.App;
+
```

Comment:
For readability, I would first bail out if this is the same region, and then continue the flow:

```ts
if (action.region === pipelineRegion) {
  return; // same region
}

// continue
```

Reply:
Changed.